### DISCUSSION TITLE
Anyone else using 64mp Owlsight Cam?

### DISCUSSION THREAD

**Discussion by jshall77** [aaronwmorris/indi-allsky#1435] [Q&A]
I've been using Aaron's version of Indi-Allsky for over a year now and have an RPi4 running a RPiCam v3 that's pretty self-sufficient.  I recently built another unit with a RPi5 and the Arducam 64mp OwlSight cam.  I'm having some issues at night where the Pi can't process images fast enough and the image queue just keeps getting further and further behind and eventually the Pi5 runs out of resources and is basically unusable.  I'm still digging through logs, but just wondering if anyone else is using the 64mp Owlsight or Hawkeye without any issues?
Thanks,
-j

**Comment by aaronwmorris** [comment]
I have the 64mp hawkeye, which I ordered just to verify functionality.  **It was challenging**

If you run the camera in RAW mode, the first problem is the size of the DNG raw file.  I believe it is 122MB.  The disk I/O with the sdcard was so slow, it was taking 15s just to write the file to disk.  The file is written to the /tmp filesystem, I even tried switching to a tmpfs (ram drive) and it was still taking 7-8 seconds just to download the file from the camera.  You could feasibly just switch to JPEG/PNG mode to make the files smaller.

The second problem is the resolution of the files presents a challenge for some of the processing algorithms like star detection and line detection.  My suggestion is just disable these entirely.

Can you send me the output from `libcamera-hello --list-cameras`?  I would like to see the resolutions available.

**Reply by jshall77** [reply]
Aaron,
As always, thank you for your quick replies!  Not only have you written a very nice application, you back up your work with your quick responses even when it's something not urgent!

Anyway, back to my question... I was wondering if the write to disk was the bottleneck.  I'm running an RPi5 with an M.2 hat and the normal performance is awesome, but I think it's trying to write too much, too fast.  I tried just changing to JPG and it doesn't make a difference, and currently have the libcamera mode set to 4624x3472, but again, it's still bottlenecked and backed up over 15 minutes.

Here's the camera info:
@hallsky:~ $ rpicam-hello --list-cameras
Available cameras
-----------------
0 : ov64a40 [9248x6944 10-bit] (/base/axi/pcie@120000/rp1/i2c@88000/ov64a40@36)
    Modes: 'SBGGR10_CSI2P' : 1920x1080 [70.56 fps - (784, 1312)/7712x4352 crop]
                             2312x1736 [32.24 fps - (0, 0)/9280x6976 crop]
                             3840x2160 [19.40 fps - (784, 1312)/7712x4352 crop]
                             4624x3472 [9.86 fps - (0, 0)/9280x6976 crop]
                             8000x6000 [3.18 fps - (624, 472)/8048x6032 crop]
                             9248x6944 [2.60 fps - (0, 0)/9280x6976 crop]

Thanks again,
-jeff

**Reply by aaronwmorris** [reply]
It is probably the star and line detection algorithms.  Check the logs to see how much time these tasks are taking.  It should be logged.

You can also show the output of the `Image Lag` view.  The database records a few statistics on how much time is being spent on parts of the image pipeline.
